src/marshaller/HTML.js

```diff
@@ -80,48 +80,67 @@
     HTML.prototype.render = function (callback) {
         if (this.ddlUrl() === "" || (this.ddlUrl() === this._prev_ddlUrl && this.databomb() === this._prev_databomb)) {
             return Tabbed.prototype.render.apply(this, arguments);
+        } else if (this._prev_ddlUrl && this._prev_ddlUrl !== this.ddlUrl()) {
+            //  DDL has actually changed (not just a deserialization)
+            this
+                .clearTabs()
+            ;
         }
         this._prev_ddlUrl = this.ddlUrl();
         this._prev_databomb = this.databomb();
 
-        this.marshaller = new HipieDDL.Marshaller().proxyMappings(this.proxyMappings());
+        //  Gather existing widgets for reuse  ---
+        var tabContent = [];
+        this.widgets().forEach(function (d) {
+            tabContent = tabContent.concat(d.widget().content());
```

Comment:
@GordonSmith Uncaught TypeError: d.widget(...).content is not a function (for DSP opening second dashboard)